[DOC] Fix make_pipeline, make_reduction,  window_summarizer & load_forecasting data docstrings

hazrulakmal opened the pull request:
#### What does this implement/fix? Explain your changes.
<!--
A clear and concise description of what you have implemented.
-->

Stumbled upon by chance a weird display on the code example snippet for `make_pipeline`  
![image](https://github.com/sktime/sktime/assets/24774385/cfab7095-91a3-4b81-bc6a-ad8a0099a7d6)

I suppose the problem lies in the indentation of the words "Example 1" and so on. However, I cannot verify it yet because it takes time to build the website locally.

Also, make some small changes to `load_forecastingdata` docstring to reflect the default option for **return_type**

yarnabrina commented:
Same issue as #5013.

yarnabrina reviewed (changes requested):
I can no longer see any comment or header specifying Example 1. Can you add that back? This may be RST feature, so may be an additional newline before will fix it.

https://sktime--5065.org.readthedocs.build/en/5065/api_reference/auto_generated/sktime.pipeline.make_pipeline.html#make-pipeline

hazrulakmal commented:
> This may be RST feature, so may be an additional newline before will fix it.

Unfortunately, pydocstyle complaints as follows:

```
sktime/pipeline/_make_pipeline.py:8 in public function `make_pipeline`:
D412: No blank lines allowed between a section header and its content ('Examples')
```

Instead, I rearranged the imports. 

Side note: 
There are others problematic code snippets. Some that I stumbled upon are
1.  [ColumnEnsembleForecaster](https://www.sktime.net/en/stable/api_reference/auto_generated/sktime.forecasting.compose.ColumnEnsembleForecaster.html#sktime.forecasting.compose.ColumnEnsembleForecaster)
2. [ForecastX](https://www.sktime.net/en/stable/api_reference/auto_generated/sktime.forecasting.compose.ForecastX.html#sktime.forecasting.compose.ForecastX)
3. [Permute](https://www.sktime.net/en/stable/api_reference/auto_generated/sktime.forecasting.compose.Permute.html#sktime.forecasting.compose.Permute)

yarnabrina commented:
Hi @hazrulakmal, I just noticed that [`make_reduction`](https://www.sktime.net/en/stable/api_reference/auto_generated/sktime.forecasting.compose.make_reduction.html) and [`WindowSummarizer`](https://www.sktime.net/en/stable/api_reference/auto_generated/sktime.transformations.series.summarize.WindowSummarizer.html) docstrings really look bad, especially where the graphical representation is not being shown as expected (I guess?). Do you want to update those in this PR as well?

hazrulakmal commented:
For [make_reduction](https://www.sktime.net/en/stable/api_reference/auto_generated/sktime.forecasting.compose.make_reduction.html) and [WindowSummarizer](https://www.sktime.net/en/stable/api_reference/auto_generated/sktime.transformations.series.summarize.WindowSummarizer.html), to ensure that the structure is maintained, they need to have double colons `:` so that their graphical representation is rendered as a code snippet.

I have also cleaned up the evaluate docstring to ensure proper rendering on documentation, and moved the graphical explanation to the notes section to meet the convention specified in the developer guide [here].(https://www.sktime.net/en/stable/developer_guide/documentation.html)

Small note: I think it is essential to display the correct format in the documentation to signal to users that sktime is well-maintained. This is usually the first stop for newcomers looking into the package. Since there is no automation available yet to check weird displays, reviewers are encouraged to check the docstring display on docs before merging a PR.

hazrulakmal commented on estimator_overview_table.md:
@fkiraly @yarnabrina Do you have any idea why this file exists in sktime? Should we delete it? It seems like it was accidentally added from https://github.com/sktime/sktime/pull/4010

yarnabrina replied:
I don't know, looks unnecessary (completely empty).

hazrulakmal replied:
this PR deleted the file. Let me know if this is not desired.

fkiraly replied:
it must have been an accident, in earlier days we manually curated the table with the estimators. This is now automatically generated, and the md used to point there. Given that it has now been empty for a while, it seems completely fine to remove it.

hazrulakmal commented:
In a nutshell, this PR fixes

1. bad graphical explanation displayed on `[WindowSummarizer](https://www.sktime.net/en/stable/api_reference/auto_generated/sktime.transformations.series.summarize.WindowSummarizer.html) and [make_reduction](https://www.sktime.net/en/stable/api_reference/auto_generated/sktime.forecasting.compose.make_reduction.html)
2. Fix typo on `load_forecasting` default input
3. remove unnecessary imports of load airline from `make_pipeline` example code snippet. 


@fkiraly @yarnabrina kindly review this and merge it is ok.

fkiraly commented on sktime/forecasting/model_evaluation/_functions.py:
```diff
@@ -320,23 +285,67 @@ def evaluate(
+
+    Notes
+    -----
+    The experiment run is the following:
```
Personally, I would leave this in the preamble. Why move this down?

hazrulakmal replied:
Resolved here https://github.com/sktime/sktime/pull/5065/commits/2e31622058ae29448613333495b6d9143d41201a

fkiraly commented on sktime/forecasting/compose/_reduce.py:
```diff
@@ -1352,72 +1352,83 @@ def make_reduction(
         (False: Number of windows = total observations + 1 - window_length
         - forecasting horizon). See pictionary below for more information.
 
-    Please see below a graphical representation of the make_reduction logic using the
```
I agree this is the wrong place, but I would actually move it up, not down, to the preamble.

hazrulakmal replied:
Resolved here https://github.com/sktime/sktime/pull/5065/commits/2e31622058ae29448613333495b6d9143d41201a

fkiraly reviewed (changes requested):
Great, thanks!

My only comment - up for discussion - is to move methodological description not to the "notes" section, but to the first paragraph, the explanation block. That's where people usually look for methodological explanation afaik.

(specific actions being: in `evaluate`, leave it be where it is; in `make_reduction`, move it up, not down)

hazrulakmal commented:
my reason for moving both down to notes follows what is stated in the sktime and numpydoc documentation convention which we should follow. This is in regard to *the extended summary section* (as you refer to *preamble*) Lemme quote one by one. 

from [sktime developer doc](https://www.sktime.net/en/stable/developer_guide/documentation.html), the extended summary section stated
> For all sktime estimators and other code artifacts that implement an algorithm (e.g. performance metrics), the extended summary should include **a short**, user-friendly synopsis of the algorithm being implemented

and from [numpydoc extended summary section](https://numpydoc.readthedocs.io/en/latest/format.html#extended-summary), I quote

> A few sentences giving an extended description. This section should be used to clarify functionality, not to discuss implementation detail or background theory, which should rather be explored in the [Notes](https://numpydoc.readthedocs.io/en/latest/format.html#notes) section below. You may refer to the parameters and the function name, but parameter descriptions still belong in the [Parameters](https://numpydoc.readthedocs.io/en/latest/format.html#params) section.

Since the graphical explanation is both not short (more than a few sentences) and discusses the implementation detail, I figured the right place for them is in the notes section.

fkiraly commented:
> Since the graphical explanation is both not short (more than a few sentences) and discusses the implementation detail, I figured the right place for them is in the notes section.

I see - I would differ in the appraisal of the content - it is not implementation detail, but a methodological summary, i.e., it describes what the estimator does.

As per the `numpydoc`, the "what does the estimator do" should be explained in the "extended summary" section.

For `evaluate`, I think the section is close to shortest possible to explain what the function does.

For `make_reduction`, I think the section is longer than it need to be and also incomplete in some aspects, but still without it the user would have no information of what the estimator does.

hazrulakmal commented:
Ok. I will move them up in the next commit.

hazrulakmal commented:
@fkiraly @yarnabrina good to go!

fkiraly reviewed (approved):
Thanks! Good from my side, the comment on the methodological description is addressed.

yarnabrina commented on sktime/pipeline/_make_pipeline.py:
```diff
@@ -20,25 +20,20 @@ def make_pipeline(*steps):
-    >>> from sktime.classification.distance_based import KNeighborsTimeSeriesClassifier
     >>> from sktime.pipeline import make_pipeline
     >>> from sktime.transformations.series.exponent import ExponentTransformer
+    >>> from sktime.classification.distance_based import KNeighborsTimeSeriesClassifier
```
I'm not sure what effect does it have. At least by `isort`, previous seemed better style. Does it affect the hosted docs?

hazrulakmal replied:
It was done so that it is consistent across all other in the docstring examples. following this steps -> import make_pipeline, then import exponenttransformer then import estimators.

yarnabrina commented on sktime/transformations/series/summarize.py:
```diff
@@ -45,45 +45,50 @@ class WindowSummarizer(BaseTransformer):
-        key: either custom function call (to be
-                provided by user) or str corresponding to native pandas window function:
+        key: either custom function call (to be provided by user) or
+            str corresponding to native pandas window function:
```
The comment is actually against line 109, but Github doesn't allow me to comment on unchanged line. SHouldn't `truncate` be one level higher than `key` or `second_value`?

hazrulakmal replied:
Yup, I agree. The previous author may have missed the indentation correction. anything else you think `window_summarizer`'s docstring should be edited?

yarnabrina replied:
No, nothing else that I can think of right now.

yarnabrina commented on sktime/transformations/series/summarize.py:
```diff
@@ -17,7 +17,7 @@
 
 class WindowSummarizer(BaseTransformer):
-    """Transformer for extracting time series features.
+    r"""Transformer for extracting time series features.
```
I'm probably missing something, but where exactly do we need the raw string?

Is it for the URL link? If it's only for that, you can take a look at `statsforecast` docstrings.

hazrulakmal replied:
see explanation below

yarnabrina commented on sktime/forecasting/compose/_reduce.py:
```diff
@@ -1310,13 +1310,83 @@ def make_reduction(
     windows_identical=True,
 ):
-    """Make forecaster based on reduction to tabular or time-series regression.
+    r"""Make forecaster based on reduction to tabular or time-series regression.
```
Just curious again, what is the raw string for?

hazrulakmal replied:
Same. see explanation below

yarnabrina reviewed (approved):
Made some comments, but IMO almost all of them can be considered non-blocking. Approved.

Also a request, can you explain what does `::` do extra over `:`, and where is it documented?

hazrulakmal commented:
> Also a request, can you explain what does :: do extra over :, and where is it documented?

I followed conventions as per `SlidingWindowSplitter` and `ExpandingWindowSplitter` which use the same graphical explanation in the docstring. Both double colon `::` and raw string `r` exit there so I assume they help sphinx recognise that the following line should be a code block so that the graphical explanation renders correctly. 

I tried a bunch of other stuff that were documented in numpydoc but to no avail. Luckily, I was also working on those two splitters and saw how their docstrings were written.
